src/server/lib/renderer.tsx

```diff
@@ -67,9 +67,21 @@ const clientStateFromRequestStateLocals = ({
         fieldErrors,
       },
     };
-  } else {
-    return clientState;
   }
+
+  // checking if recaptcha error exists in query params, and attaching it to the
+  // forms field errors
+  if (queryParams.recaptchaError) {
+    return {
+      ...clientState,
+      globalMessage: {
+        ...clientState.globalMessage,
+        error: CaptchaErrors.GENERIC,
+      },
```

Comment:
can we use `deepmerge` here as we do elsewhere in the project?

Reply:
switched to using `deepmerge`